adapta/utils/README.md

```diff
@@ -107,4 +107,44 @@ with memory_limit(memory_limit_percentage=0.2):
     # Do some stuff here
     # if the provided limit is exceeded, MemoryError will be raised and can be caught by client code
     pass
+```
+
+### runtime_metrics:
+
+Decorator for logging method execution time and reporting function execution time to metric provider (DataDog).  
+
+Notes:
+* Function call is logged with loglevel.INFO
+* Function runtime is logged with loglevel.DEBUG
+
+Parameters:
+* method_type (optional): Type of method being wrapped. This will be used as a metric_name by the metric provider.
+* pass_logger (optional): If True, the logger will be passed to the wrapped function.
+* pass_metrics_provider (optional): If True, the metrics provider will be passed to the wrapped function.
+* extra_metric_entities (optional): Dictionary containing extra tags used by the metric provider.
+
+Example Usage:
+```python
+import time
+from adapta.metrics.providers.datadog_provider import DatadogMetricsProvider
+from adapta.logs import SemanticLogger
+from adapta.logs.models import LogLevel
+from adapta.utils import run_time_metrics
+
+
+@run_time_metrics(method_type='example', pass_logger=True, pass_metrics_provider=True)
```

Comment:
I think this can also pass logger and metrics arguments regardless if the function uses them internally?